### PR TITLE
Added Quantize option

### PR DESCRIPTION
**Pull request by yusufcakmakk**
Merging existing Onnx inference with Quantize option.
This request is based Huggingface [nootbeook ](https://github.com/huggingface/transformers/blob/master/notebooks/04-onnx-export.ipynb). Also Microsoft has similar [notebook](https://github.com/microsoft/onnxruntime/blob/master/onnxruntime/python/tools/quantization/notebooks/Bert-GLUE_OnnxRuntime_quantization.ipynb).

### CODE REVIEW

**Comment by patil-suraj** [comment]
Thank you for this awesome feature @yusufcakmakk !

Could you send the PR from a different branch than master ? 
This needs few changes, we could do it on that branch and then merge.

File: onnx_transformers/pipelines.py [hunk 1]
```diff
@@ -555,7 +559,18 @@ def __init__(
             logger.info(f"loading onnx graph from {self.graph_path.as_posix()}")
             self.onnx_model = create_model_for_provider(str(graph_path), "CPUExecutionProvider")
             self.input_names = json.load(open(input_names_path))
-            self.framework = "np"
+            self.framework = "pt"
```
Comment (patil-suraj) [inline onnx_transformers/pipelines.py:562]: better to keep it `np`, `onnx` requires inputs as `np` arrays, so converting to `torch` and then back to `np` seems unnecessary.

Is there any reason you made this change ?

File: onnx_transformers/pipelines.py [hunk 2]
```diff
@@ -555,7 +559,18 @@ def __init__(
...
+            if quantized:
+                onnx_opt_model_path = graph_path.parent.joinpath(f"{graph_path.stem}-opt.onnx")
+                quantized_model_path = graph_path.parent.joinpath(f"{graph_path.stem}-opt-quantized.onnx")
+                if not quantized_model_path.exists() or not onnx_opt_model_path.exists():
+                    opt_options = BertOptimizationOptions('bert')
```
Comment (patil-suraj) [inline onnx_transformers/pipelines.py]: we could move this logic to separate method. Will be cleaner.

File: onnx_transformers/pipelines.py [hunk 3]
```diff
@@ -555,7 +559,18 @@ def __init__(
...
+                    opt_options.enable_embed_layer_norm = False
+                    onnx_opt_model = optimizer.optimize_model(graph_path.as_posix(),'bert',num_heads=12,hidden_size=768,optimization_options=opt_options)
```
Comment (patil-suraj) [inline onnx_transformers/pipelines.py]: we should read `num_heads`, `hidden_size` from `config`, hard coding this will break it for large models

**Comment by patil-suraj** [comment]
Let me know if you are busy, I will work on it and include you as co-author :)

Thanks again!

File: onnx_transformers/pipelines.py (hunk 1 above)
Comment (yusufcakmakk) [reply onnx_transformers/pipelines.py:562]: In some cases I've got error that `InvalidArgument: [ONNXRuntimeError] : 2 : INVALID_ARGUMENT : Unexpected input data type.` When I changed it to "pt", it is solved. That is why I changed it to torch.

**Comment by yusufcakmakk** [comment]
I will try to update as you [mentioned](https://github.com/patil-suraj/onnx_transformers/pull/5#issuecomment-706537911). By the way you are welcome. I hope it will be useful 😊

File: onnx_transformers/pipelines.py (hunk 1 above)
Comment (yusufcakmakk) [reply onnx_transformers/pipelines.py:562]: What do you think of this case? Should i keep it as np? @patil-suraj

**Pull request closed without merging**